test-app/config/application.rb

```diff
@@ -0,0 +1,72 @@
+require_relative "boot"
+
+require "rails"
+# Pick the frameworks you want:
+require "active_model/railtie"
+# require "active_job/railtie"
+require "active_record/railtie"
+# require "active_storage/engine"
+require "action_controller/railtie"
+# require "action_mailer/railtie"
+# require "action_mailbox/engine"
+# require "action_text/engine"
+require "action_view/railtie"
+# require "action_cable/engine"
+require "rails/test_unit/railtie"
+
+# Require the gems listed in Gemfile, including any gems
+# you've limited to :test, :development, or :production.
+Bundler.require(*Rails.groups)
+
+module OverWriteMetrics
+  def record_sampling_metrics
+    @metrics[:tracecount].add(rand(1..10))
+    @metrics[:samplecount].add(rand(1..10))
+    @metrics[:request_count].add(rand(1..10))
+    @metrics[:toex_count].add(rand(1..10))
+    @metrics[:through_count].add(rand(1..10))
+    @metrics[:tt_count].add(rand(1..10))
+  end
+
+  # new metrics sdk use periodic table by default
+  def on_finish(span)
+    SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] processor on_finish span: #{span.to_span_data.inspect}" }
+
+    return if non_entry_span(span: span)
+
+    record_request_metrics(span)
+    record_sampling_metrics
+
+    SolarWindsAPM.logger.debug { "[#{self.class}/#{__method__}] processor on_finish succeed" }
+  rescue StandardError => e
+    SolarWindsAPM.logger.info { "[#{self.class}/#{__method__}] error processing span on_finish: #{e.message}" }
+  end
+end
+
+require 'solarwinds_apm'
+
+SolarWindsAPM::OpenTelemetry::OTLPProcessor.prepend(OverWriteMetrics) if defined? (::SolarWindsAPM::OpenTelemetry)
```

Comment:
Same comment as above, that we should remove all of this cusomization and expect it to be supported in the RC gem.

Reply:
Removed; and use the build from [github package](https://github.com/users/xuan-cao-swi/packages/rubygems/solarwinds_apm/333257165)